samcli/local/docker/container.py

```diff
@@ -266,8 +271,13 @@ def wait_for_http_response(self, name, event, stdout):
         # TODO(sriram-mv): `aws-lambda-rie` is in a mode where the function_name is always "function"
         # NOTE(sriram-mv): There is a connection timeout set on the http call to `aws-lambda-rie`, however there is not
         # a read time out for the response received from the server.
+
+        # if localhost is not set, we assume it's running on the local machine.
+        if not self._localhost:
+            self._localhost = "localhost"
```

Comment:
The default should be set at the CLI level (where the `click.option` is), so that it shows up with `--help`.